apiserver/common/modelmanagerinterface.go

```diff
@@ -111,6 +112,22 @@ type CloudService interface {
 	WatchCloud(ctx context.Context, name string) (watcher.NotifyWatcher, error)
 }
 
+// MachineService defines the methods that the facade assumes from the Machine
+// service.
+type MachineService interface {
+	// EnsureDeadMachine sets the provided machine's life status to Dead.
+	// No error is returned if the provided machine doesn't exist, just nothing
+	// gets updated.
+	EnsureDeadMachine(ctx context.Context, machineName machine.Name) error
+	// GetMachineUUID returns the UUID of a machine identified by its name.
+	// It returns a MachineNotFound if the machine does not exist.
```

Comment:
This part of the comment was correctly removed because errors are not part of the interface since it provides implementation details. They *could* be part of the comment describing a method implementation maybe, but not the interface.

Reply:
I disagree on that as long as some code relies on this error. 

It's unfortunate to not have a way to enforce it by some way in the signature, but since there is some code that rely on the fact that this method return a specific error, the specific error is part of the contract and should be documented as well, and implementation should stick with it (and not returning any other error for the same meaning).